modules/mysql/main.tf

```diff
@@ -258,6 +264,19 @@ resource "google_sql_user" "additional_users" {
   ]
 }
 
+resource "google_sql_user" "iam_account" {
+  for_each = local.iam_users
+
+  project  = var.project_id
+  name     = each.value.email
```

Comment:
If this change works, I am confused why it didn't also break the postgresql module:

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/e638a24eacc953d3e8ebf248c12bda0b66021745/modules/postgresql/main.tf#L266C13-L274

Reply:
Perhaps it might be worth re-running the test-suite for the postgresql and checking this:
https://github.com/terraform-google-modules/terraform-google-sql-db/blob/e638a24eacc953d3e8ebf248c12bda0b66021745/examples/postgresql-public-iam/main.tf#L72-L76